dataikuapi/dss/ml.py

```diff
@@ -315,43 +313,979 @@ def save(self):
+
+    def __init__(self, name, algo_settings, accepted_values=None):
+        super(SingleCategoryHyperparameterSettings, self).__init__(name, algo_settings)
+        self.accepted_values = accepted_values
+
+    def __repr__(self):
+        if self.accepted_values is not None:
+            return self.__class__.__name__ + "(hyperparameter=\"{}\", value=\"{}\", accepted_values={})".format(self.name,
+                                                                                                            self._algo_settings[self.name],
+                                                                                                            self.accepted_values)
+        else:
+            return self.__class__.__name__ + "(hyperparameter=\"{}\", value=\"{}\")".format(self.name, self._algo_settings[self.name])
+
+    __str__ = __repr__
+
+    _pretty_repr = __repr__
+
+    def set_value(self, value):
+        """
+        :param value:
+        :type value: str
+        """
+        if self.accepted_values is not None:
+            assert value in self.accepted_values, "Invalid value for hyperparameter {}. Must be in {}".format(self.name, json.dumps(self.accepted_values))
+        self._algo_settings[self.name] = value
+
+    def get_value(self):
+        """
+        :return: current value
+        :rtype: str
+        """
+        return self._algo_settings[self.name]
+
+    def get_all_possible_values(self):
+        """
+        :return: list of possible values for this hyperparameter
+        :rtype: list of str
+        """
+        return self.accepted_values
+
+
+class PredictionAlgorithmSettings(dict):
+    """
+    Object to read and modify the settings of a prediction ML algorithm.
+
+    Do not create this object directly, use :meth:`DSSMLTask.get_algorithm_settings(algorithm)` instead
+    """
+
+    def __init__(self, raw_settings, hyperparameter_search_params):
+        super(PredictionAlgorithmSettings, self).__init__(raw_settings)
+        self._hyperparameter_search_params = hyperparameter_search_params
+        self._hyperparameters_registry = dict()
+
+    def __setattr__(self, key, value):
+        if not hasattr(self, key):
+            # call from __init__
+            super(PredictionAlgorithmSettings, self).__setattr__(key, value)
+        elif key in self._hyperparameters_registry:
+            # syntactic sugars
+            target = self._hyperparameters_registry[key]
+            if isinstance(target, (SingleValueHyperparameterSettings, SingleCategoryHyperparameterSettings)):
+                target.set_value(value)
+            elif isinstance(target, CategoricalHyperparameterSettings):
+                target.set_values(value)
+            else:
+                raise Exception("Invalid assignment of a NumericalHyperparameterSettings object")
+        elif key == "lambda_":
+            raise Exception("Invalid assignment of a NumericalHyperparameterSettings object")
```

Comment:
Why not set the `.values`? Not sure why `algo.cat = ["v1", "v2"]` would work but not `algo.num = [1, 2]`?

Reply:
I agree 👍 
Should we just set `.values` or `set_explicit_values` for `NumericalHyperparameterSettings`  ?

IMO should be `set_explicit_values` for the syntactic sugars, so that the user does not wonder why "nothing happens after setting values"

Reply:
> I disagree: having syntactic sugar for explicit values makes them look
privileged compared to ranges, and we would likely have it look the other
way around, especially to advanced users.

In that case it would be nice to have some syntactic sugar for both explicit and range values. Something like:
* If the value is a `list` => `set_explicit_values`
* if the value is a `Range` => `set_range`

But this would require to modify the `Range` class so that it is independent of the `NumericalHyperparameterSettings`. 

Not sure if it is worth it

Reply:
OK let's say that we can bench this for now, and not yet have direct assignment for multi-valued numerical HPs.

Reply:
Created [CH58941](https://app.clubhouse.io/dataiku/story/58941/public-api-python-client-add-syntactic-sugar-for-numericalhyperparametersettings) to this purpose.